### ISSUE TITLE
Unable to publisth event from node

### ISSUE THREAD

**Issue by ddysher** [GoogleCloudPlatform/kubernetes#4379]
```
Namespace error: invalid value '': event.Namespace default, involvedObject.Namespace ' (will not retry!)
```

event namespace is different than node namespace, even though event namespace is set to node namespace.  We are still having problems with node namespace.  I'll take a look.

**Comment by ddysher** [comment]
The problem is actually multi-level.

tr;dr  We need to fix event namespace to handle NamespaceNone

In apiserver, when a resource is created without namespace in path or in query param, we set it to api.NamespaceDefault.  This happens at URL parse time, before touching RESTStorage.  It is then passed down using context object.

When control goes to node RESTStorage, the create stategy says node shouldn't have namespace, and namespace is explicitly set to api.NamespaceNone.

When publishing event from node, the enclosing namespace should be api.NamespaceNone. And each individual event is set to the node's namespace, which is api.NamespaceNone.  This is expected, but client will *not* construct namespace param/path if namespace is None.  So apiserver will parse it to api.NamespaceDefault as mentioned above.  This is where the above validation error comes in.

A temp solution is to clear the namespace before validation.  I don't have a better solution off my head now.

**Comment by ddysher** [comment]
We can also fix it by equiping event rest with create strategy.

**Comment by ddysher** [comment]
The issue and PR #4396 is blocked on node namespace.

**Comment by ddysher** [comment]
This is no longer relevant now i think. node is recording event with no namespace.
